crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs

```diff
@@ -0,0 +1,218 @@
+use crate::{
+    keywords::{WEBKIT_SCROLLBAR_PSEUDO_CLASSES, WEBKIT_SCROLLBAR_PSEUDO_ELEMENTS},
+    utils::{is_custom_selector, is_known_pseudo_class, is_page_pseudo_class, vendor_prefixed},
+};
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{
+    CssBogusPseudoClass, CssPageSelectorPseudo, CssPseudoClassFunctionCompoundSelector,
+    CssPseudoClassFunctionCompoundSelectorList, CssPseudoClassFunctionIdentifier,
+    CssPseudoClassFunctionNth, CssPseudoClassFunctionRelativeSelectorList,
+    CssPseudoClassFunctionSelector, CssPseudoClassFunctionSelectorList,
+    CssPseudoClassFunctionValueList, CssPseudoClassIdentifier, CssPseudoElementSelector,
+};
+use biome_rowan::{declare_node_union, AstNode, SyntaxNodeCast, TextRange};
+
+declare_rule! {
+    /// Disallow unknown pseudo-class selectors.
+    ///
+    /// For details on known pseudo-class, see the [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes)
+    ///
+    /// This rule ignores vendor-prefixed pseudo-class selectors.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// a:unknown {}
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// a:UNKNOWN {}
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// a:hoverr {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// a:hover {}
+    /// ```
+    ///
+    /// ```css
+    /// a:focus {}
+    /// ```
+    ///
+    /// ```css
+    /// :not(p) {}
+    /// ```
+    ///
+    /// ```css
+    /// input:-moz-placeholder {}
+    /// ```
+    ///
+    pub NoUnknownPseudoClassSelector {
+        version: "next",
+        name: "noUnknownPseudoClassSelector",
+        language: "css",
+        recommended: false,
+        sources: &[RuleSource::Stylelint("selector-pseudo-class-no-unknown")],
+    }
+}
+declare_node_union! {
+    pub AnyPseudoLike = CssPseudoClassFunctionCompoundSelector|CssPseudoClassFunctionCompoundSelectorList|CssPseudoClassFunctionIdentifier
+                        |CssPseudoClassFunctionNth|CssPseudoClassFunctionRelativeSelectorList|CssPseudoClassFunctionSelector
+                        |CssPseudoClassFunctionSelectorList|CssPseudoClassFunctionValueList|CssPseudoClassIdentifier
+                        |CssBogusPseudoClass|CssPageSelectorPseudo
+}
+
+fn is_webkit_pseudo_class(node: &AnyPseudoLike) -> bool {
+    let mut prev_element = node.syntax().parent().and_then(|p| p.prev_sibling());
+    while let Some(prev) = &prev_element {
+        let maybe_selector: Option<CssPseudoElementSelector> = prev.clone().cast();
+        if let Some(selector) = maybe_selector.as_ref() {
+            return WEBKIT_SCROLLBAR_PSEUDO_ELEMENTS.contains(&selector.text().trim_matches(':'));
+        };
+        prev_element = prev.prev_sibling();
+    }
+
+    false
+}
```

Comment:
To determine whether a `::-webkit-scrollbar CSS pseudo-element` has a selector, I'm retrieving each previous element one by one, but I feel there might be a performance issue.